bridges/zap/zap_encoder.go

```diff
@@ -0,0 +1,211 @@
+package zap
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"go.opentelemetry.io/otel/log"
+	"go.uber.org/zap/zapcore"
+)
+
+// not optimized yet
+// this file implements object and array encoder - similar to memory encoder by zapcore
+var (
+	_ zapcore.ObjectEncoder = (*OtelObjectEncoder)(nil)
+	_ zapcore.ArrayEncoder  = (*sliceArrayEncoder)(nil)
+)
+
+type OtelObjectEncoder struct {
+	// Fields contains the entire encoded log context.
+	Fields []log.KeyValue
+	// cur is a pointer to the namespace we're currently writing to.
+	cur []log.KeyValue
+
+	reflectval log.Value
+	zapcore.Encoder
```

Comment:
Why is this embedded? If there are methods this is providing, when they are called by a user it will cause a panic. We should implement all the needed methods instead.